src/UI/home/dashboard/chart.tsx

```diff
@@ -1,20 +1,51 @@
-import {PieChart} from "react-minimal-pie-chart";
-import React from "react";
-import {cryptoChart} from "../types";
+import {PieChart} from 'react-minimal-pie-chart'
+import React, {useEffect, useState} from 'react'
+import {cryptoChart} from '../types'
+import {CoinInWallet} from '../../../backend-service-connector/model/rest'
 
 export function Chart(props: any) {
-    const userService = props.userService
+    const userWallet = props.userWallet
+    const allUserCoins: Array<CoinInWallet> = userWallet.allCoinsInWallet
+    const [color, setColor] = useState<Array<any>>([])
+    let randomColor = require('randomcolor')
 
-    const data: Array<cryptoChart> = userService.getDataForChart()
+    useEffect(() => {
+        let arrayOfColors: Array<any> = []
+
+        for (let i = 0; i < allUserCoins.length; i++) {
+            const color = randomColor()
+
+            arrayOfColors = arrayOfColors.concat(color)
+        }
```

Comment:
that can be simplified to:
`
const arrayOfColors = allUserCoins.map(() => randomColor())
`